src/genie/libs/parser/iosxe/show_snmp.py

```diff
@@ -80,6 +81,329 @@ def cli(self, output=None):
+        p_output_count = re.compile(r"^(?P<output_count>\d+)\s+SNMP\s+packets\s+output$")
+
+        # 0 Too big errors (Maximum packet size 1500)
+        p_output_big = re.compile(r"^(?P<output_big>\d+)\s+Too\s+big\s+errors\s+\(Maximum\s+packet\s+size\s+1500\)$")
+
+        # 1 No such name errors
+        p_output_name = re.compile(r"^(?P<output_name>\d+)\s+No\s+such\s+name\s+errors$")
+
+        # 0 Bad values errors
+        p_output_bad = re.compile(r"^(?P<output_bad>\d+)\s+Bad\s+values\s+errors$")
+
+        # 0 General errors
+        p_output_general = re.compile(r"^(?P<output_general>\d+)\s+General\s+errors$")
+
+        # 322899 Response PDUs
+        p_output_response_pdu = re.compile(r"^(?P<output_response_pdu>\d+)\s+Response\s+PDUs$")
+
+        # 527788 Trap PDUs
+        p_output_trap_pdu = re.compile(r"^(?P<output_trap>\d+)\s+Trap\s+PDUs")
+
+        # Packets currently in SNMP process input queue: 0
+        p_packet_queue = re.compile(r"^Packets\s+currently\s+in\s+SNMP\s+process\s+input\s+queue:\s+(?P<queue>\d+)")
+
+        # SNMP global trap: enabled
+        p_global_trap = re.compile(r"^SNMP\s+global\s+trap:\s+enabled")
+
+        # SNMP logging: enabled
+        p_logging = re.compile(r"^SNMP\s+logging:\s+(?P<logging_status>(enabled|disabled))$")
+
+        # Logging to 13.43.1.12.151, 0/10, 102045 sent, 3065 dropped.
+        p_logging_endpoint = re.compile(r"^Logging\s+to\s+(?P<endpoint_ip>\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\.(?P<endpoint_port>\d+),\s+(?P<endpoint_buffer>\d+\/\d+),\s+(?P<endpoint_sent>\d+)\s+sent,\s+(?P<endpoint_dropped>\d+)\s+dropped.$")
+
+        snmp_obj = {}
+
+        for line in output.splitlines():
+            line = line.strip()
+            # Chassis: FCE2310A48M
+            if p_chassis.match(line):
+                match = p_chassis.match(line)
+                snmp_obj["chassis"] = match.group("chassis")
+                continue
+            # Contact: "something@cisco.com"
+            elif p_contact.match(line):
+                line = line.replace('"', '')
+                match = p_contact.match(line)
+                snmp_obj["contact"] = match.group("contact")
+                continue
+            # Location: "To be filled in"
+            elif p_location_not_filled.match(line):
+                match = p_location_not_filled.match(line)
+                snmp_obj["location"] = "To be filled in"
+                continue
+            # Location: SJCT
+            elif p_location.match(line):
+                match = p_location.match(line)
+                snmp_obj["location"] = match.group("location")
+                continue
+            # 35072424 SNMP packets input
+            elif p_input_count.match(line):
+                match = p_input_count.match(line)
+                if not snmp_obj.get("snmp_input"):
+                    snmp_obj["snmp_input"] = {}
+                snmp_obj["snmp_input"].update({ "packet_count": int(match.group("input_count"))})
```

Comment:
Any access to `snmp_obj["snmp_input"]` or `snmp_obj["snmp_output"]` could be done as 
```suggestion
                snmp_obj.setdefault("snmp_input", {})["packet_count"] = int(match.group("input_count"))
```
This would be more robust and prevent any possibility of key errors, since `setdefault` will return any existing dict, or create a new empty one.

Reply:
This is a neat feature. I will keep it in mind going forward. Thanks. File updated.